utils/build/build.mk

```diff
@@ -0,0 +1,75 @@
+# This makefile configures variables related to compilation, e.g. compiler flags.
+# Furthermore, it defines compilation recipes
+CFLAGS   := $(CFLAGS) -pthread -Wall
+LFLAGS   := $(LFLAGS) -lpthread -pthread
+CC       := g++
+INC_DIR  := $(INC_DIR) -I$(SRCS_DIR) -I$(LIBS_DIR)
+DEPFLAGS  = -MT $@ -MMD -MP -MF $(OBJS_DIR)/$*.d
```

Comment:
```suggestion
DEPFLAGS  = -MT $@ -MMD -MP -MF $(OBJS_DIR)/$*.d
# DEPFLAGS ensures that changes to headers trigger recompilation properly. More info on the wiki:
# https://github.com/Hyp-ed/hyped-2020/wiki/Makefiles#what-is-the-line-depflags----mt---mmd--mp--mf-objs_dird-in-buildmk
```

Reply:
Btw, what is the policy regarding small changes like this? I think you could have committed the comments yourself directly. Are only the PR authors allowed to make changes? And reviewers are only able to suggest?